File: Python/Trees and Graphs/Q5.py
from BaseTreeImplementation import Node
# Implement a function to check if a binary tree is a binary search tree

temp = []


def is_binary_search_tree(root):
    in_order(root)
    for i in xrange(len(temp)):
        if i + 1 < len(temp):
            if temp[i] > temp[i + 1]:
                return False
    return True


def in_order(root):
    if root.left is not None:
        in_order(root.left)

    temp.append(root.data)

    if root.right is not None:
        in_order(root.right)


def main():
    root = Node(10)
    root.left = Node(9)
    root.right = Node(5)

    print is_binary_search_tree(root)
    print

if __name__ == '__main__':
    main()

File: Java/Hired Programming Practice/src/TreesAndGraphs/Q5.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package TreesAndGraphs;
import java.util.ArrayList;
/**
 *
 * @author marcsantiago
 */

//Implement a function to check if a binary tree is a binary search tree
public class Q5 {
       
    ArrayList<Integer> list = new ArrayList<>();
    public boolean isBinarySearchTree(Node root){
        inorder(root);
        for (int i = 0; i < list.size(); i++) {
            if(i + 1 < list.size()){
                if(list.get(i) > list.get(i + 1)){
                    return false;
                }
            }
        }
        return true;
    }
    
    public void inorder(Node root){
        if(root.leftPointer != null){
            inorder(root.leftPointer);
        }
        list.add(root.data);
        if(root.rightPointer != null){
            inorder(root.rightPointer);
        }
    }
    
    
}
